supervisor/supervisor.go

```diff
@@ -0,0 +1,171 @@
+package supervisor
```

Comment:
Since supervisor.go is not used in Windows, so is it possible to add `// +build !windows` and add dummy supervisor_windows.go?
(I don't know the best way. One might think current way is better because of simplicity).